adrs/0001-cli-design.md

```diff
@@ -0,0 +1,306 @@
+# 1. CLI design
+
+Date: 2023-01-06
+
+## Status
+
+Draft
+
+## Context
+
+Red Hat Product Security provides tooling supporting users security related queries - 
+relaying information on security related entities such as vulnerabilities (CVEs/weaknesses)
+and affected products and components.
+
+The CLI that integrates with security related data services will provide:
+
+* Set of core entity operations on flaws, affects, components, products, etc... for 
+searching, listing and retrieving entities.
+* Set of read only query operations answering 'canned' product security related queries
+* Set of process operations (mutation) automating away manual 'drudgery'
+* Dynamic, extensible set of custom plugin operations for interacting with external services 
+* Separate area for management operations of underlying prodsec services (ex. OSIDB, CORGI) 
+
+The CLI provides a simple 'facade' over coarse grained security related data services allowing 
+for easier aggregation and narrowing of information providing a good security 'signal' 
+for end users.
+
+## CLI Interface Design
+
+The command line interface is navigated as a hierarchichal tree:
+
+```commandline
+griffon/
+├─ docs/
+├─ entities/
+│  ├─ flaws
+│  ├─ affects
+│  ├─ trackers
+│  ├─ components
+│  ├─ product_versions
+│  ├─ product_streams
+│  ├─ channels
+│  ├─ **TBD**
+├─ manage/
+│  ├─ osidb
+│  ├─ corgi
+├─ queries/
+│  ├─ component_cves
+│  ├─ components_affected_by_cve
+│  ├─ components_in_product_stream
+│  ├─ cves_for_product_version
+│  ├─ product_versions_affected_by_cve
+│  ├─ **TBD**
+├─ process/
+│  ├─ generate_affects_for_component
+│  ├─ **TBD**
+```
```

Comment:
I guess that there would be also something like core for the basic functionality shared between the managed modules? May be argument parsing or some settings etc...

Reply:
maybe it is better to add top level 'service' verb and have process/queries underneath that ... this more aligns with 'service layer' ... agreed internally we can reuse options